tools/wptrunner/wptrunner/wptrunner.py

```diff
@@ -162,106 +162,113 @@ def get_pause_after_test(test_loader, **kwargs):
+
+    for test_type in kwargs["test_types"]:
+        for subsuite_name, subsuite in test_loader.subsuites.items():
+            tests_by_type[(subsuite_name, test_type)].extend(test_loader.tests[subsuite_name][test_type])
+            tests_by_type[(subsuite_name, test_type)].extend(test_loader.disabled_tests[subsuite_name][test_type])
 
-    test_implementation_by_type = {}
+    tests_by_group = test_source.cls.tests_by_group(tests_by_type, **kwargs)
 
-    for test_type in run_test_kwargs["test_types"]:
+    for test_type in kwargs["test_types"]:
         executor_cls = product.executor_classes.get(test_type)
         if executor_cls is None:
             logger.warning(f"Unsupported test type {test_type} for product {product.name}")
             continue
-        executor_kwargs = product.get_executor_kwargs(logger,
-                                                      test_type,
-                                                      test_environment,
-                                                      run_info,
-                                                      **run_test_kwargs)
         browser_cls = product.get_browser_cls(test_type)
-        browser_kwargs = product.get_browser_kwargs(logger,
-                                                    test_type,
-                                                    run_info,
-                                                    config=test_environment.config,
-                                                    num_test_groups=len(test_groups),
-                                                    **run_test_kwargs)
-        test_implementation_by_type[test_type] = TestImplementation(executor_cls,
-                                                                    executor_kwargs,
-                                                                    browser_cls,
-                                                                    browser_kwargs)
 
-    tests_to_run = {}
-    for test_type, test_implementation in test_implementation_by_type.items():
-        executor_cls = test_implementation.executor_cls
-
-        for test in test_loader.disabled_tests[test_type]:
-            logger.test_start(test.id)
-            logger.test_end(test.id, status="SKIP")
-            test_status.skipped += 1
-
-        if test_type == "testharness":
-            tests_to_run[test_type] = []
-            for test in test_loader.tests[test_type]:
-                if ((test.testdriver and not executor_cls.supports_testdriver) or
-                        (test.jsshell and not executor_cls.supports_jsshell)):
-                    logger.test_start(test.id)
-                    logger.test_end(test.id, status="SKIP")
-                    test_status.skipped += 1
-                else:
-                    tests_to_run[test_type].append(test)
-        else:
-            tests_to_run[test_type] = test_loader.tests[test_type]
+        for subsuite_name, subsuite in test_loader.subsuites.items():
```

Comment:
We think we need to loop through subsuite first, then test_type. We need a restart for each subsuite, but not necessary for a test type change.